Documentation/ExtensionArchitecture/Extbase/Reference/_FrontendPlugin/_ExternalRedirectController.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+// Example by: https://brot.krue.ml/extbase-controller-action-responses-in-typo3/
+final class MyController extends ActionController
+{
+    public function downloadAction(): ResponseInterface
+    {
+        // ... do something (set $value, ...)
+
+        $uri = $this->uriBuilder->uriFor('show', ['parameter' => $value]);
+
+        // $uri could also be https://example.com/any/uri
+        return $this->responseFactory->createResponse(307)
```

Comment:
add $responseFactory as class variable. Then we can also see what type is have. Or does it come from the Action controller? then I would leave a comment

Reply:
Yes, `responseFactory` is injected via `ActionController`. What kind of comment would you like? Something like:

```
// $this->resourceFactory is injected as part of the `ActionController` inheritance
```

?

Reply:
(Added in the comment, feel free to rephrase)

Reply:
Fine for me :-)